### PR TITLE
ROCm 3.5.1 image

### PR DESCRIPTION
**Pull request by jeffdaily**

### CODE REVIEW

**Comment by jeffdaily** [comment]
@ailzhang @ezyang Does the documentation for building circleci images need updating? 

https://github.com/pytorch/pytorch/wiki/Docker-image-build-on-CircleCI

The Lint check is complaining about config.yaml consistency because the trigger was removed.  Removing it directly from the config.yaml seemed better than removing the code that generates the trigger: https://github.com/pytorch/pytorch/blob/87c5f02f3dccd67b61bdb5098c6048f5981dc1e2/.circleci/generate_config_yml.py#L120-L126

Note though that the lint check will pass once the config.yaml is restored after all images have been built.

**Comment by ezyang** [comment]
Yes, looks like those docs need updating. If you remove directly from config.yaml this will indeed cause the consistency check to fail. But this should be benign, you'll still build the Docker images anyway; then you can revert it and we can land it.

**Review by malfet — APPROVED** [review]

**Comment by jeffdaily** [comment]
@sunway513 @pruthvistony please review these changes.  Notably, the sccache wrapping of the HIP compilers under /opt/rocm has changed, and the unwrap script as been removed since it is no longer needed.

@sunway513 the miopenkernels are also installed, as requested.

@malfet please confirm your requests are part of the change set.  Notably, the sccache support and kmod install.

**Comment by jeffdaily** [comment]
@ezyang @malfet I'm not seeing all images uploaded and indexed.  The last time we created a new rocm image, the new tag was `be76e8fd-44e2-484d-b090-
07e0cc3a56f0` and there are 18 images with that tag.  But for this PR, the tag is `cdd4d248cdc589ccfce5c999d8163f1292902f5d` but there are only 9 images with that tag.  Did something go wrong?

**Comment by ezyang** [comment]
The last build you did was https://app.circleci.com/pipelines/github/pytorch/pytorch/185416/workflows/02a63234-f7c5-4ae0-a472-e5e2be15eca6 which looks like everything except the android build succeeded

**Comment by ezyang** [comment]
cc @seemethere looking at https://app.circleci.com/pipelines/github/pytorch/pytorch/185416/workflows/02a63234-f7c5-4ae0-a472-e5e2be15eca6/jobs/6016915 it looks like some of the jobs are uploading to the wrong hash. Sounds like a bug in #40194.

**Comment by sunway513** [comment]
cc @ashishfarmer to follow up with this PR while @jeffdaily  is on vacation.

File: .circleci/generate_config_yml.py [hunk 1]
```diff
@@ -117,14 +117,14 @@ def gen_build_workflows_tree():
...
-                        }
-                    ],
+                    #"triggers": [
+                    #    {
+                    #        "schedule": {
+                    #            "cron": miniutils.quote("0 15 * * 0"),
+                    #            "filters": {"branches": {"only": ["master"]}},
+                    #        }
+                    #    }
+                    #],
```
Comment (malfet) [inline .circleci/generate_config_yml.py:127]: @jeffdaily you probably don't want to comment that out, do you?

**Review by malfet — APPROVED** [review]
@jeffdaily can you please rebase this PR (and keep schedule triggers for docker build)

**Review by seemethere — COMMENTED** [review]
LGTM, aside from removing the cron scheduler

File: .circleci/generate_config_yml.py (hunk 1 above)
Comment (jeffdaily) [reply .circleci/generate_config_yml.py:127]: The trigger will be restored once all images have successfully built.

**Comment by jeffdaily** [comment]
Do we still have a problem?  Looks like the "Calculate docker tag" for these builds is giving `DOCKER_TAG=3efc1f019f3affeb83b23d0020c84ccf1d0b95b2`, but some are uploading with the tag `ab1632df-fa59-40e6-8c23-98e004f61148` while others upload with the calculated docker tag.

**Comment by jeffdaily** [comment]
@seemethere we still have two distinct docker tags used.  Should I just add both tags to the don't-GC list?

**Comment by jeffdaily** [comment]
All 24 images built, divided between two tags.

**Comment by seemethere** [comment]
They're actually being pushed with both tags right now, either one should be fine to use but in the future we're planning on going with the one that's based on the `git rev-parse HEAD:.circleci/docker` output.

For now you can use either one.

File: .circleci/generate_config_yml.py (hunk 1 above)
Comment (seemethere) [reply .circleci/generate_config_yml.py:127]: This should be fine to uncomment out now that the images have been built

**Comment by jeffdaily** [comment]
@seemethere I'm going to save the one that's showing up in the http://docker.pytorch.org/pytorch.html index for the rocm images.  `ab1632df-fa59-40e6-8c23-98e004f61148`.

**Review by ezyang — APPROVED** [review]

**Pull request closed without merging**